src/gui/passkeys/PasskeyImportDialog.cpp

```diff
@@ -0,0 +1,121 @@
+#include "ui_PasskeyImportDialog.h"
+
+#include "browser/BrowserService.h"
+#include "core/Metadata.h"
+#include "gui/MainWindow.h"
+#include <QCloseEvent>
+#include <QFileInfo>
+
+PasskeyImportDialog::PasskeyImportDialog(QWidget* parent)
+    : QDialog(parent)
+    , m_ui(new Ui::PasskeyImportDialog())
+    , m_useDefaultGroup(true)
+{
+    setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
+
+    m_ui->setupUi(this);
+    m_ui->useDefaultGroupCheckbox->setChecked(true);
+    m_ui->selectGroupComboBox->setEnabled(false);
+
+    connect(m_ui->importButton, SIGNAL(clicked()), SLOT(accept()));
+    connect(m_ui->cancelButton, SIGNAL(clicked()), SLOT(reject()));
+    connect(m_ui->selectDatabaseButton, SIGNAL(clicked()), SLOT(selectDatabase()));
+    connect(m_ui->selectGroupComboBox, SIGNAL(currentIndexChanged(int)), SLOT(changeGroup(int)));
+    connect(m_ui->useDefaultGroupCheckbox, SIGNAL(stateChanged(int)), SLOT(useDefaultGroupChanged()));
+}
+
+PasskeyImportDialog::~PasskeyImportDialog()
+{
+}
+
+void PasskeyImportDialog::setInfo(const QString& url, const QString& username, const QSharedPointer<Database>& database)
+{
+    m_ui->urlLabel->setText(QString("%1: %2").arg(tr("URL"), url));
+    m_ui->usernameLabel->setText(QString("%1: %2").arg(tr("Username"), username));
+    m_ui->selectDatabaseLabel->setText(QString("%1: %2").arg(tr("Database"), getDatabaseName(database)));
+    m_ui->selectGroupLabel->setText(QString("%1:").arg(tr("Group")));
+
+    addGroups(database);
+
+    auto openDatabaseCount = 0;
+    for (auto dbWidget : getMainWindow()->getOpenDatabases()) {
+        if (dbWidget && !dbWidget->isLocked()) {
+            openDatabaseCount++;
+        }
+    }
+    m_ui->selectDatabaseButton->setEnabled(openDatabaseCount > 1);
+}
+
+QSharedPointer<Database> PasskeyImportDialog::getSelectedDatabase()
+{
+    return m_selectedDatabase;
+}
+
+QUuid PasskeyImportDialog::getSelectedGroupUuid()
+{
+    return m_selectedGroupUuid;
+}
+
+bool PasskeyImportDialog::useDefaultGroup()
+{
+    return m_useDefaultGroup;
+}
+
+QString PasskeyImportDialog::getDatabaseName(const QSharedPointer<Database>& database) const
+{
+    return QFileInfo(database->filePath()).fileName();
+}
+
+void PasskeyImportDialog::addGroups(const QSharedPointer<Database>& database)
+{
+    m_ui->selectGroupComboBox->clear();
+    for (const auto& group : database->rootGroup()->groupsRecursive(true)) {
+        if (!group || group->isRecycled() || group == database->metadata()->recycleBin()) {
+            continue;
+        }
+
+        m_ui->selectGroupComboBox->addItem(group->fullPath(), group->uuid());
+    }
+}
+
+void PasskeyImportDialog::selectDatabase()
+{
+    auto selectedDatabase = browserService()->selectedDatabase();
+    if (!selectedDatabase) {
+        return;
+    }
+
+    m_selectedDatabase = selectedDatabase;
+    m_ui->selectDatabaseLabel->setText(QString("%1: %2").arg(tr("Database"), getDatabaseName(m_selectedDatabase)));
```

Comment:
```suggestion
    m_ui->selectDatabaseLabel->setText(tr("Database: %1").arg(getDatabaseName(m_selectedDatabase)));
```

Reply:
(There might be more of these which I may have missed while reading through the code)